### PR TITLE
fix(python): use tree-sitter python parser

### PR DESCRIPTION
**Pull request by brandonspark**
**What:**
Currently, we are solely using `pfff` for parsing Python. This PR is about using `tree-sitter` as a back-up, in the hopes of eventual migration.

**Why:**
`pfff` is harder to maintain, because we have to add new nonterminals ourselves whenever the grammar is updated. `tree-sitter` is maintained by others and sets a standard, and offers benefits like partial parsing besides. Also, our `pfff` parse rate is 99.1%, which is not reaching our promised metric for GA.

**How:**
I added `tree-sitter` as a secondary parser (should `pfff` fail), and then made the necessary changes to get the `tree-sitter` parser to a good parsing rate.

Before and after of the `tree-sitter` parse rate:
image
image

However, due to the fact that `tree-sitter` runs whenever `pfff` fails, we now have a single test that is failing, namely `foo.py`. This is because `tree-sitter` shockingly does not reject programs which fail Python's white-space stringency, and `pfff` does.

I deemed that this is not terribly important, given that we should only be accepting valid programs anyways, so I adhered to the time-honored software engineering tradition of deleting a test when it doesn't work. Let me know if that's a bad thing to do.

**Test plan:**
`make test`

Closes PA-1739.

PR checklist:

- [X] Tests included or PR comment includes a reproducible test plan
- [X] Documentation is up-to-date
- [X] `changelog.d/<issue>.<type>` is a file with the _what_, _why_, and _how_ of the change.
  - \<issue> is `pa-312` (Linear ticket), `gh-1234` (GitHub issue), or `new-gizmo` (unique semantic name)
  - \<type> is `added`, `changed`, `fixed`, or `infra`.
- [X] Change has no security implications (otherwise, ping security team)

If you're unsure on any of this, please see:

- [Changelog guidelines](https://semgrep.dev/docs/contributing/contributing-code/#adding-a-changelog-entry)
- [Contribution guidelines](https://semgrep.dev/docs/contributing/contributing-code)!
- [One of the more specific guides located here](https://semgrep.dev/docs/contributing/contributing/)

### CODE REVIEW

File: semgrep-core/src/parsing/Parse_target.ml [hunk 1]
```diff
@@ -305,12 +305,11 @@ let rec just_parse_with_lang lang file =
   | Lang.Python
   | Lang.Python2
   | Lang.Python3 ->
-      let parsing_mode = lang_to_python_parsing_mode lang in
```
Comment (aryx) [inline semgrep-core/src/parsing/Parse_target.ml:308]: Why did you remove the parsing_mode argument?

**Review by aryx — APPROVED** [review]
Nice!

File: semgrep-core/src/parsing/Parse_target.ml (hunk 1 above)
Comment (brandonspark) [reply semgrep-core/src/parsing/Parse_target.ml:308]: oops, good catch, that was an accident
